### PR TITLE
Add RagAgentDocGrader  to agent comp

### PR DESCRIPTION
**Pull request by minmin-intel**
## Description

Add a new agent strategy for agentic RAG

## Issues

 `n/a`.

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds new functionality)

## Dependencies

Need OpenAI LLM for difficult queries. Simple queries can be solved by open-source LLMs.

## Tests

Unit test in GenAIComps/tests/

### CODE REVIEW

**Comment by minmin-intel** [comment]
@xuechendi @XuhuiRen Please review. Thanks!

File: comps/agent/langchain/src/strategy/docgrader/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Agentic Rag
```
Comment (xuechendi) [inline comps/agent/langchain/src/strategy/docgrader/README.md:1]: this readme is not updated

**Comment by xuechendi** [comment]
The PR looks good to me, I think we should replace agentic Rag with DocGrader instead of creating a new one. These two looks really similiar, and seems DocGrader is also working on non-openAI API, so do we still need to keep both?

File: comps/agent/langchain/src/strategy/docgrader/README.md (hunk 1 above)
Comment (minmin-intel) [reply comps/agent/langchain/src/strategy/docgrader/README.md]: updated readme.

**Comment by minmin-intel** [comment]
@xuechendi Deleted the old agentic_rag strategy

**Review by xuechendi — APPROVED** [review]
LGTM

File: comps/agent/langchain/src/strategy/docgrader/planner.py [hunk 2]
```diff
@@ -4,76 +4,31 @@
...
 from ..base_agent import BaseAgent
-from .prompt import rlm_rag_prompt
+from .prompt import DOC_GRADER_PROMPT, RAGv1_PROMPT
 
+instruction = "Retrieved document is not sufficient or relevant to answer the query. Reformulate the query to search knowledge base again."
+MAX_RETRY = 3
 
-class AgentState(TypedDict):
+
+class AgentStateV1(TypedDict):
```
Comment (XuhuiRen) [inline comps/agent/langchain/src/strategy/docgrader/planner.py:25]: why all the name of the classes have a suffix "V1"?

**Comment by XuhuiRen** [comment]
what is the difference between "TextGrader" and "DocumentGrader"? should be better give some introduction in the readme

File: comps/agent/langchain/src/strategy/docgrader/planner.py (hunk 2 above)
Comment (minmin-intel) [reply comps/agent/langchain/src/strategy/docgrader/planner.py:25]: Removed V1

**Comment by minmin-intel** [comment]
@XuhuiRen I added description of DocumentGrader and TextGenerator in the readme in the docgrader subfolder.

File: tests/test_agent_langchain.sh [hunk 3]
```diff
@@ -69,38 +94,69 @@ function validate() {
...
+    start_tgi_service
+
+    # test react_langchain
+    start_react_langchain_agent_service
+    echo "=============Testing ReAct Langchain============="
+    validate_microservice
+    stop_agent_docker
+    echo "============================================="
+
+    # # test react_langgraph
```
Comment (chensuyue) [inline tests/test_agent_langchain.sh:147]: why comment those test?

Comment (minmin-intel) [reply tests/test_agent_langchain.sh:147]: This test needs to use OpenAI model GPT-4o-mini. @chensuyue

**Comment by minmin-intel** [comment]
Tests with OpenAI LLMs passed on my system. See screen shot below. @chensuyue Let me know if this is sufficient, or you would want me to test OpenAI LLM again in CICD.
image

**Comment by minmin-intel** [comment]
@XuhuiRen have changed the docgrader to rag_agent, and I have updated the overall README and the README in the strategy/ragagent/ subfolder, describing the rag agent. The plan is to iterate upon the ragagent strategy if we develop more performant rag agents in future.

**Comment by chensuyue** [comment]
> Tests with OpenAI LLMs passed on my system. See screen shot below. @chensuyue Let me know if this is sufficient, or you would want me to test OpenAI LLM again in CICD. image

To make sure the function always work, we must add the test into CI/CD, in case it is impact by other PRs.

**Review by XuhuiRen — APPROVED** [review]

**Pull request merged**